components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -315,6 +316,79 @@ std::vector<mojom::FeedItemV2Ptr> GenerateChannelBlock(
   return result;
 }
 
+std::vector<mojom::FeedItemV2Ptr> GenerateTopicBlock(
+    std::vector<Topic>& topics,
+    const Publishers& publishers) {
+  if (topics.empty()) {
+    return {};
+  }
+  DVLOG(1) << __FUNCTION__;
+  auto result = mojom::Cluster::New();
+  result->type = "topic";
+
+  auto topic = std::move(topics[0]);
+  topics.erase(topics.begin());
```

Comment:
Q. This seems to result in popping out the first element from `topic_` member variable. Is this intended? This feels like this could be a potential pitfall.

Reply:
Yup, this is intended - we take a copy of the topics before we start building the feed. We remove the `Topics` from the vector as we use them, so we don't show the same topics multiple times.

Reply:
Aha! Thanks for the explanation. That makes sense.